### ISSUE TITLE
相关文件中torch.load()不兼容torch2.6.0以上的版本

### ISSUE THREAD

**Issue by HX958** [ultralytics/yolov5#13675]
### Search before asking

- [x] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and found no similar bug report.

### Bug

在v7.0版本中，为啥很多代码文件中，对torch 2.6.0以上的版本都不兼容，刚下载的代码库，很多文件中只要有torch.cuda.amp.autocast()就会报错，然后还得一个一个修改，太麻烦了。这些文件有train.py、val.py、autobatch.py、torch_utils.py。有torch.load()的也会报错，这些文件有general.py、train.py、resume.py、hubconf.py、experimental.py、yolov5/utils/loggers/__init__.py，对于Pillow 10.0.0+以上的版本，有PIL.ImageFont.load_default().getsize()也会报错，只能用getbbox()，这些文件有plots.py

### Are you willing to submit a PR?

- [x] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @HX958, thank you for your interest in YOLOv5 🚀! This is an automated response to help you get unblocked quickly—an Ultralytics engineer will also assist you here soon.

From your report, it sounds like you’re encountering issues with newer PyTorch and Pillow versions (e.g., torch 2.6+ around autocast/torch.load, and Pillow 10+ with font sizing). If this is a 🐛 Bug Report, please provide a minimum reproducible example (MRE) to help us debug it effectively:
- Exact commands you ran and a small sample to reproduce
- Full error traceback(s)
- Your environment details: OS, Python, PyTorch, CUDA, CUDNN, Pillow versions
- The specific YOLOv5 commit or release tag (e.g., v7.0)

If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our Tips for Best Training Results.

## Requirements
[**Python>=3.8.0**](https://www.python.org/) with all [requirements.txt](https://github.com/ultralytics/yolov5/blob/master/requirements.txt) installed including [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/). To get started:
```bash
git clone https://github.com/ultralytics/yolov5  # clone
cd yolov5
pip install -r requirements.txt  # install
```

## Environments
YOLOv5 may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):
- **Notebooks** with free GPU: <a href="https://bit.ly/yolov5-paperspace-notebook"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"></a> <a href="https://colab.research.google.com/github/ultralytics/yolov5/blob/master/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolov5"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/yolov5"><img src="https://img.shields.io/docker/pulls/ultralytics/yolov5?logo=docker" alt="Docker Pulls"></a>

## Status
<a href="https://github.com/ultralytics/yolov5/actions/workflows/ci-testing.yml"><img src="https://github.com/ultralytics/yolov5/actions/workflows/ci-testing.yml/badge.svg" alt="YOLOv5 CI"></a>
If this badge is green, all [YOLOv5 GitHub Actions](https://github.com/ultralytics/yolov5/actions) Continuous Integration (CI) tests are currently passing. CI tests verify correct operation of YOLOv5 [training](https://github.com/ultralytics/yolov5/blob/master/train.py), [validation](https://github.com/ultralytics/yolov5/blob/master/val.py), [inference](https://github.com/ultralytics/yolov5/blob/master/detect.py), [export](https://github.com/ultralytics/yolov5/blob/master/export.py) and [benchmarks](https://github.com/ultralytics/yolov5/blob/master/benchmarks.py) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

Since you’re open to submitting a PR ✅, that’s greatly appreciated! Please include:
- The MRE demonstrating the issue
- The exact files and lines impacted (e.g., train.py, val.py, general.py, plots.py)
- Backwards compatibility notes and minimal, targeted changes

Thanks again for the detailed heads-up—once we have your MRE and environment details, we can help investigate swiftly 🙏

**Comment by pderrenger** [comment]
感谢反馈！v7.0 是 2022 年的旧标签（见我们的 YOLOv5 v7.0 公告），与 PyTorch 2.6+/Pillow 10+ 的 API 变更确实会不兼容；请先在当前 master 拉取并按 requirements 重新安装后再试，若仍复现：将 torch.cuda.amp.autocast() 替换为 torch.autocast('cuda', dtype=torch.float16)，对 torch.load(...) 显式传 weights_only=False（旧版可 try/except 回退），并把 ImageFont.getsize() 改为 font.getbbox(text) 后用 bbox 计算宽高。你已勾选愿意提 PR，欢迎基于 master 提交这些兼容修改，我们会尽快评审；提交流程见 CONTRIBUTING。临时规避可先 pip install "torch<2.6" "pillow<10"。需要环境与安装说明可参考 Ultralytics Docs；若希望直接获得对最新依赖的支持，也可考虑使用 Ultralytics YOLO（YOLO11）。参见 YOLOv5 v7.0 公告和 Ultralytics Docs。
